src/main/java/net/dv8tion/jda/core/entities/impl/InviteImpl.java

```diff
@@ -303,7 +304,7 @@ public GuildImpl(final long id, final String iconId, final String name, final St
             this.verificationLevel = verificationLevel;
             this.presenceCount = presenceCount;
             this.memberCount = memberCount;
-            this.features = features;
+            this.features = Collections.unmodifiableSet(features);
```

Comment:
You are possibly wrapping the `Collections.emptySet()` in an unmodifiable decorator for no reason. Instead you should do this when you actually create a set.